training_model/one_file_train.py

```
@@ -440,96 +1651,270 @@ def copy_data(
+
+    Raises:
+        DataProcessingError: If dataset upload fails.
+    """
+    _load_environment_if_needed()
+    password = os.getenv("PASSWORD_BOT")
+    try:
+        url = "https://dataset.ser13volk.me/dataset_ru"
+        dataset_path = Path("../data") / "dataset_ru.json"
+
+        if not dataset_path.exists():
+            raise DataProcessingError(f"Dataset file not found: {dataset_path}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)